src/libvirtApi/common.js

```diff
@@ -161,88 +161,91 @@ function doUsagePolling(name, connectionName, objPath) {
+                props.rssMemory = 0.0;
+            for (let i = 0; i < info["vcpu.maximum"].v.v; i++) {
+                if (!(`vcpu.${i}.time` in info))
+                    continue;
+                avgvCpuTime += info[`vcpu.${i}.time`].v.v;
+            }
+            avgvCpuTime /= info["vcpu.current"].v.v;
+            if (info["vcpu.current"].v.v > 0)
+                Object.assign(props, {
+                    actualTimeInMs: Date.now(),
+                    cpuTime: avgvCpuTime
+                });
+            Object.assign(props, {
+                disksStats: calculateDiskStats(info)
+            });
+
+            logDebug(`doUsagePolling: ${JSON.stringify(props)}`);
+            store.dispatch(updateVm(props));
+        }
+    } catch (ex) {
+        console.warn(`GetStats(${name}, ${connectionName}) failed: ${ex.toString()}`);
+    }
+    delayPolling(() => doUsagePolling(name, connectionName, objPath), null);
 }
 
-function getLoggedInUser() {
-    logDebug(`GET_LOGGED_IN_USER:`);
-    return cockpit.user().then(loggedUser => store.dispatch(setLoggedInUser({ loggedUser })));
+async function getLoggedInUser() {
+    const loggedUser = await cockpit.user();
+    logDebug(`GET_LOGGED_IN_USER:`, loggedUser);
+    store.dispatch(setLoggedInUser({ loggedUser }));
 }
 
-export function getLibvirtVersion({ connectionName }) {
-    return call(connectionName, "/org/libvirt/QEMU", "org.freedesktop.DBus.Properties", "Get", ["org.libvirt.Connect", "LibVersion"], { timeout, type: "ss" })
-            .then(version => store.dispatch(updateLibvirtVersion({ libvirtVersion: version[0].v })));
+export async function getLibvirtVersion({ connectionName }) {
+    const [version] = await call(connectionName, "/org/libvirt/QEMU", "org.freedesktop.DBus.Properties", "Get", ["org.libvirt.Connect", "LibVersion"], { timeout, type: "ss" });
+    store.dispatch(updateLibvirtVersion({ libvirtVersion: version.v }));
 }
 
-function getNodeMaxMemory({ connectionName }) {
+async function getNodeMaxMemory({ connectionName }) {
     // Some nodes don"t return all memory in just one cell.
     // Using -1 == VIR_NODE_MEMORY_STATS_ALL_CELLS will return memory across all cells
-    return call(connectionName, "/org/libvirt/QEMU", "org.libvirt.Connect", "NodeGetMemoryStats", [-1, 0], { timeout, type: "iu" })
-            .then(stats => store.dispatch(setNodeMaxMemory({ memory: stats[0].total })))
-            .catch(ex => {
-                console.warn("NodeGetMemoryStats failed:", ex.toString());
-                return Promise.reject(ex);
-            });
+    try {
+        const [stats] = await call(connectionName, "/org/libvirt/QEMU", "org.libvirt.Connect", "NodeGetMemoryStats", [-1, 0], { timeout, type: "iu" });
+        store.dispatch(setNodeMaxMemory({ memory: stats.total }));
+    } catch (ex) {
+        console.warn("NodeGetMemoryStats failed:", ex.toString());
+        throw ex;
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1400-20240125-053137-bfbdf98b-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)